From ad526ed010333ffcac566a5a6bc43589fd389e82 Mon Sep 17 00:00:00 2001
From: Ralf Gommers <ralf.gommers@gmail.com>
Date: Sat, 8 Apr 2023 22:45:37 +0100
Subject: [PATCH 4/9] MAINT: remove `from numpy.math cimport` usages

We'd like to get rid of `libnpymath`, and the (unrelated)
`numpy.math` in the Python API was also just deprecated.
So best to not use `numpy.math` from Cython either.

[skip circle]
---
 scipy/sparse/csgraph/_matching.pyx      | 2 +-
 scipy/sparse/csgraph/_shortest_path.pyx | 2 +-
 scipy/spatial/_ckdtree.pyx              | 3 +--
 scipy/spatial/transform/_rotation.pyx   | 3 +--
 scipy/special/_ellip_harm.pxd           | 3 +--
 scipy/special/_hyp0f1.pxd               | 3 +--
 scipy/stats/_stats.pyx                  | 5 ++---
 7 files changed, 8 insertions(+), 13 deletions(-)

diff --git a/scipy/sparse/csgraph/_matching.pyx b/scipy/sparse/csgraph/_matching.pyx
index a92e8a91e..e1ed15d77 100644
--- a/scipy/sparse/csgraph/_matching.pyx
+++ b/scipy/sparse/csgraph/_matching.pyx
@@ -3,7 +3,7 @@ import warnings
 cimport cython
 import numpy as np
 cimport numpy as np
-from numpy.math cimport INFINITY
+from libc.math cimport INFINITY
 
 
 from scipy.sparse import isspmatrix_coo, isspmatrix_csc, isspmatrix_csr
diff --git a/scipy/sparse/csgraph/_shortest_path.pyx b/scipy/sparse/csgraph/_shortest_path.pyx
index 459672ca2..680b4d4a7 100644
--- a/scipy/sparse/csgraph/_shortest_path.pyx
+++ b/scipy/sparse/csgraph/_shortest_path.pyx
@@ -20,7 +20,7 @@ from scipy.sparse.csgraph._validation import validate_graph
 cimport cython
 
 from libc.stdlib cimport malloc, free
-from numpy.math cimport INFINITY
+from libc.math cimport INFINITY
 
 np.import_array()
 
diff --git a/scipy/spatial/_ckdtree.pyx b/scipy/spatial/_ckdtree.pyx
index aaa641eee..eadfdef6f 100644
--- a/scipy/spatial/_ckdtree.pyx
+++ b/scipy/spatial/_ckdtree.pyx
@@ -10,12 +10,11 @@ import numpy as np
 import scipy.sparse
 
 cimport numpy as np
-from numpy.math cimport INFINITY
 
 from cpython.mem cimport PyMem_Malloc, PyMem_Free
 from libcpp.vector cimport vector
 from libcpp cimport bool
-from libc.math cimport isinf
+from libc.math cimport isinf, INFINITY
 
 cimport cython
 import os
diff --git a/scipy/spatial/transform/_rotation.pyx b/scipy/spatial/transform/_rotation.pyx
index d97a9b0cd..86ae595e8 100644
--- a/scipy/spatial/transform/_rotation.pyx
+++ b/scipy/spatial/transform/_rotation.pyx
@@ -7,8 +7,7 @@ from ._rotation_groups import create_group
 cimport numpy as np
 cimport cython
 from cython.view cimport array
-from libc.math cimport sqrt, sin, cos, atan2, acos
-from numpy.math cimport PI as pi, NAN, isnan # avoid MSVC error
+from libc.math cimport sqrt, sin, cos, atan2, acos, hypot, isnan, NAN, pi
 
 np.import_array()
 
diff --git a/scipy/special/_ellip_harm.pxd b/scipy/special/_ellip_harm.pxd
index 4f83d05c3..431c6a017 100644
--- a/scipy/special/_ellip_harm.pxd
+++ b/scipy/special/_ellip_harm.pxd
@@ -32,9 +32,8 @@ import cython
 
 from . cimport sf_error
 
-from libc.math cimport sqrt, fabs, pow
+from libc.math cimport sqrt, fabs, pow, NAN
 from libc.stdlib cimport malloc, free
-from numpy.math cimport NAN, PI
 
 cdef extern from "lapack_defs.h":
     ctypedef int CBLAS_INT  # actual type defined in the header
diff --git a/scipy/special/_hyp0f1.pxd b/scipy/special/_hyp0f1.pxd
index baacca8cf..54d7374e5 100644
--- a/scipy/special/_hyp0f1.pxd
+++ b/scipy/special/_hyp0f1.pxd
@@ -1,5 +1,4 @@
-from libc.math cimport pow, sqrt, floor, log, log1p, exp, M_PI, fabs
-from numpy.math cimport NAN, isinf
+from libc.math cimport pow, sqrt, floor, log, log1p, exp, M_PI, NAN, fabs, isinf
 cimport numpy as np
 
 from ._cephes cimport iv, jv, Gamma, lgam, gammasgn
diff --git a/scipy/stats/_stats.pyx b/scipy/stats/_stats.pyx
index b8496b1f3..94c5f060d 100644
--- a/scipy/stats/_stats.pyx
+++ b/scipy/stats/_stats.pyx
@@ -1,11 +1,10 @@
 from cpython cimport bool
 from libc cimport math
+from libc.math cimport NAN, INFINITY, M_PI as PI
 cimport cython
 cimport numpy as np
-from numpy.math cimport PI
-from numpy.math cimport INFINITY
-from numpy.math cimport NAN
 from numpy cimport ndarray, int64_t, float64_t, intp_t
+
 import warnings
 import numpy as np
 import scipy.stats, scipy.special
